src/AppInstallerCLITests/RestInterface_1_0.cpp

```diff
@@ -0,0 +1,413 @@
+            REQUIRE(actualInstaller.Dependencies.WindowsLibraries.at(0) == "library1");
+            REQUIRE(actualInstaller.Dependencies.PackageDependencies.at(0).Id == "Foo.Baz");
+            REQUIRE(actualInstaller.Dependencies.PackageDependencies.at(0).MinVersion == "2.0.0");
+            REQUIRE(actualInstaller.Dependencies.ExternalDependencies.at(0) == "FooBarBaz");
+            REQUIRE(actualInstaller.PackageFamilyName == "FooBar.PackageFamilyName");
+            REQUIRE(actualInstaller.ProductCode == "");
+            REQUIRE(actualInstaller.Capabilities.at(0) == "bluetooth");
+            REQUIRE(actualInstaller.RestrictedCapabilities.at(0) == "restrictedCapability");
+        }
+    };
+}
+
+TEST_CASE("Search_GoodResponse", "[RestSource]")
+{
+    utility::string_t sample = _XPLATSTR(
+        R"delimeter({
+            "Data" : [
+               {
+              "PackageIdentifier": "git.package",
+              "PackageName": "package",
+              "Publisher": "git",
+              "Versions": [
+                {   "PackageVersion": "1.0.0" },
+                {   "PackageVersion": "2.0.0"}]
+            }]
+        })delimeter");
+
+    HttpClientHelper helper{ GetTestHandler(web::http::status_codes::OK, std::move(sample)) };
+    Interface v1{ TestRestUriString, std::move(helper) };
+    Schema::IRestClient::SearchResult searchResponse = v1.Search({});
+    REQUIRE(searchResponse.Matches.size() == 1);
+    Schema::IRestClient::Package package = searchResponse.Matches.at(0);
+    REQUIRE(package.PackageInformation.PackageIdentifier.compare("git.package") == 0);
+    REQUIRE(package.PackageInformation.Publisher.compare("git") == 0);
+    REQUIRE(package.PackageInformation.PackageName.compare("package") == 0);
+    REQUIRE(package.Versions.size() == 2);
+    REQUIRE(package.Versions.at(0).VersionAndChannel.GetVersion().ToString().compare("1.0.0") == 0);
+    REQUIRE(package.Versions.at(1).VersionAndChannel.GetVersion().ToString().compare("2.0.0") == 0);
+}
+
+TEST_CASE("Search_BadResponse", "[RestSource]")
+{
+    utility::string_t sample = _XPLATSTR(
+        R"delimeter({
+            "Data" : [
+               {
+              "PackageIdentifier": "git.package",
+              "PackageName": "package",
+              "Publisher": "git",
+              "Versions": null }]
+        })delimeter");
+
+    HttpClientHelper helper{ GetTestHandler(web::http::status_codes::OK, std::move(sample)) };
+    Interface v1{ TestRestUriString, std::move(helper) };
+    REQUIRE_THROWS_HR(v1.Search({}), APPINSTALLER_CLI_ERROR_RESTSOURCE_INVALID_DATA);
+}
+
+TEST_CASE("Search_BadResponse_NotFoundCode", "[RestSource]")
+{
+    HttpClientHelper helper{ GetTestHandler(web::http::status_codes::NotFound) };
+    Interface v1{ TestRestUriString, std::move(helper) };
+    Schema::IRestClient::SearchResult result = v1.Search({});
+    REQUIRE(result.Matches.empty());
+}
+
+TEST_CASE("Search_Optimized_ManifestResponse", "[RestSource]")

```

Comment:
>Optimized [](start = 18, length = 9)

what does the Optimized mean?

Reply:
We call the Rest source to give us the manifest of a package instead of searching for its availability and then getting the manifest when user asks to install a particular package. Thus this is called optimized search

---
In reply to: [612904828](https://github.com/microsoft/winget-cli/pull/861#discussion_r612904828) [](ancestors = 612904828)